db/src/tx.rs

```diff
@@ -103,6 +103,14 @@ use types::{
 };
 use upsert_resolution::Generation;
 
+struct PartRanges(Vec<(i64, i64)>);
```

Comment:
You could profitably newtype `PartitionMap`, simplifying this slightly.  Either that or add it to the `PartitionMapping` trait that already exists.